phpBB/phpbb/content_visibility.php

```diff
@@ -145,13 +145,13 @@ public function get_count($mode, $data, $forum_id)
 	public function is_visible($mode, $forum_id, $data)
 	{
 		$visibility = $data[$mode . '_visibility'];
-		$poster_key = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+
 		$is_visible = ($visibility == ITEM_APPROVED) ||
-			($this->config['display_unapproved_posts'] &&
-				($this->user->data['user_id'] != ANONYMOUS) &&
-				($visibility == ITEM_UNAPPROVED || $visibility == ITEM_REAPPROVE) &&
-				($this->user->data['user_id'] == $data[$poster_key])) ||
```

Comment:
This is needed to show unapproved posts to their authors. Hence this code incorrect.

Reply:
What I meant to say is to the post's author only (or users who can approve the post). This code looks like it will show the post to anyone who is logged in.

Reply:
Not really, try it.

I tried it with me as the founder, with the poster itself and with another tester.

Reply:
For the "another tester" a "topic does not exist" it is shown, BTW.

Reply:
BTW, we are talking about attachments here.

Reply:
Sorry for the delay, so this seem to work because the query is already filtering the user ID. However, the real issue is not here. There are two errors in the ticket:
1) `... on line 153: Undefined array key "topic_poster"`
2) `... on line 153: Undefined array key "poster_id"`

1) is comming from [here](https://github.com/phpbb/phpbb/blob/5169a947f4caef876cd2f38c558d042aef0a24ae/phpBB/includes/functions_download.php#L659-L676), note that `topic_poster` is missing from the query.
2) is comming from [here](https://github.com/phpbb/phpbb/blob/3.3.x/phpBB/download/file.php#L219-L236), note that `poster_id` is missing from the query.

Reply:
I didn't have those 2 errors, also not in the logs.

Reply:
those should be notices. I'm going by the ticket, the user said they saw those when clicking download, which makes sense. So this should error out when you go via `file.php` which should result in broken images, as the content served is text with the notices the reporter posted.

Reply:
Yeah sure, btw the attachments are those I posted there. Just I haven't had any NOTICE nor error_logs.
Wich is quite strange in my environment, I am double checking the PHP 8.0.7 configuration.